File: golang-backend/internal/model/scan_task.go
package model

type ScanTask struct {
	ID              int64   `gorm:"primaryKey;autoIncrement" json:"id"`
	Name            string  `gorm:"not null" json:"name"`
	Description     *string `json:"description"`
	TaskType        string  `gorm:"column:task_type;not null;index:idx_scan_task_type_enabled" json:"taskType"`
	ScanRangeType   string  `gorm:"column:scan_range_type;not null" json:"scanRangeType"`
	StartDate       *int64  `gorm:"column:start_date" json:"startDate"`
	EndDate         *int64  `gorm:"column:end_date" json:"endDate"`
	CronExpression  *string `gorm:"column:cron_expression" json:"cronExpression"`
	RepositoryIDs   *string `gorm:"column:repository_ids" json:"repositoryIds"`
	Enabled         bool    `gorm:"default:true;index:idx_scan_task_type_enabled;index:idx_scan_task_enabled" json:"enabled"`
	LastExecuteTime *int64  `gorm:"column:last_execute_time" json:"lastExecuteTime"`
	SortOrder       int     `gorm:"column:sort_order;default:0;index:idx_scan_task_sort_order" json:"sortOrder"`
	CreatedAt       int64   `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt       int64   `gorm:"column:updated_at" json:"updatedAt"`
}

func (ScanTask) TableName() string { return "scan_tasks" }

File: python-backend/app/models/scan_task.py
"""ScanTask ORM model."""

from sqlalchemy import BigInteger, Boolean, Index, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class ScanTask(Base):
    __tablename__ = "scan_tasks"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    description: Mapped[str | None] = mapped_column(String, nullable=True)
    task_type: Mapped[str] = mapped_column(String, nullable=False)
    scan_range_type: Mapped[str] = mapped_column(String, nullable=False)
    start_date: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    end_date: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    cron_expression: Mapped[str | None] = mapped_column(String, nullable=True)
    repository_ids: Mapped[str | None] = mapped_column(String, nullable=True)
    enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    last_execute_time: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[int] = mapped_column(BigInteger, nullable=False)
    updated_at: Mapped[int] = mapped_column(BigInteger, nullable=False)

    __table_args__ = (
        Index("idx_scan_task_type_enabled", "task_type", "enabled"),
        Index("idx_scan_task_enabled", "enabled"),
        Index("idx_scan_task_sort_order", "sort_order"),
    )
